pkg/controlplane/instance_test.go

```diff
@@ -301,7 +301,7 @@ func TestStorageVersionHashes(t *testing.T) {
 		APIPath:       "/api",
 		ContentConfig: restclient.ContentConfig{NegotiatedSerializer: legacyscheme.Codecs},
 	}
-	discover := discovery.NewDiscoveryClientForConfigOrDie(c)
+	discover := discovery.NewDiscoveryClientForConfigOrDie(c).WithLegacy()
```

Comment:
Why do we need legacy here?

Reply:
Storage version hash is omitted in aggregated discovery. This test specifically tests for the existence of storage version hash so it should continue to use the legacy discovery.

Reply:
I've added deprecating and planning for removing the StorageVersionHash to the apimachinery agenda for next meeting.